accounts-db/src/append_vec.rs

```diff
@@ -1277,13 +1277,6 @@ pub mod tests {
             }
         }
 
-        fn get_executable_byte(&self) -> u8 {
```

Comment:
This change seems unrelated to the unit tests speed up.
Why not move it into a separate PR?

Reply:
The only reason this change is happening is to make optimizations possible for unittests. For more substantial changes separate PRs have been made by teams handling respective pieces of the code. I'll be making some as well.

Reply:
If this check is not really useful (as is claimed in the discussion) it can be done in a separate PR, as it has value by itself.
Making things simpler and avoiding UB, IIUC.
And merging smaller PRs is generally easier.

For example, should you end up introducing a separate compilation profile for the CI, this PR will increase in size and will take some time to land.

But this particular change might go in pretty quickly on its own.

It is up to you, of course.  I was just speaking from personal experience :)